### DISCUSSION TITLE
Intraprocess pub/sub

### DISCUSSION THREAD

**Discussion by skldfm** [eclipse-iceoryx/iceoryx#1997] [General]
Could I use iceoryx to create (low latency) pub/sub data channels within a multithreaded application? (*intra*-process communication)

Or iceoryx is not the best choice for this task and it can be used only with limitations?

**Comment by elBoberido** [comment]
iceoryx can also be used in a single process environment. There is also an example available. Depending of what you want to achieve it might be more heavy than for example a lock-free queue.

**Reply by skldfm** [reply]
The data channels would be (highly) configurable even in run-time. I understand that shared memory access is a bit heavier than a custom  pub/sub layer above lock-free queues but I don't really want to reinvent the wheel :) . 

Thank you for your answer, I'll look into the example!
